programs/nft-club/src/components/benefit.rs

```diff
@@ -22,10 +24,18 @@ pub fn create_benefit(ctx: Context<CreateBenefit>, description: String) -> Resul
 // Data Validators
 // 
 #[derive(Accounts)]
-#[instruction(description: String)]
+#[instruction(name: String, description: String)]
 pub struct CreateBenefit<'info> {
     // Create account of type Benefit and assign creator's pubkey as the payer
-    #[account(init, payer = authority, space = Benefit::LEN)]
+    // This also makes sure that we have only one benefit for the following combination
+    // creatorPubKey + name of benefit + "benefit".
+    #[account(
+        init, 
+        seeds = [creator.key().as_ref(), name.as_ref(), b"benefit".as_ref()], 
```

Comment:
We discussed via Discord, but leaving a record here. Having the benefit name is fine, it just can't be used as a seed for the account, because the FE clients would have no way of knowing what a benefit name is until they look at the benefit account (and the seeds are required to fetch that account)